File: php/test_m2m.php
<?php

include_once('dyn-connector.php');


class test {

	private $access_token = null;

	/** Called when the connector needs last access token
	 *
	 * @return Last access token used
	 */
	public function get_access_token() {
		// Load from disk or database
		$this->access_token = @file_get_contents('dyn-connector.dat');

		echo "Connector requires access token (\"".$this->access_token."\")\n";

		return $this->access_token;
	}

	/** Called when the connector get a new access token
	 *
	 * @param string $access_token Token to save
	 */
	public function set_access_token($access_token) {
		$this->access_token = $access_token;

		echo "New access token \"".$this->access_token."\"\n";

		// Write to disk or database
		@file_put_contents('dyn-connector.dat', $this->access_token);
	}

	public function normal_request() {
		$domain = 'pigrecoos.it';

		$connector = new dyn_connector($domain);

		// Enable debugging output (should be disabled in production)
		$connector->set_debug(true);

		$data = array(
			'test' => array(
				'key1' => 1,
				'key2' => true,
				'key3' => 'This is a test'
			)
		);

		$result = $connector->send('test/echo', $data);

		echo "Server response: \n";
		print_r($result);
	}

	public function authenticated_request() {
		$domain = 'pigrecoos.it';
		$client_id = 'client ID';
		$client_secret = 'client secret';

		$connector = new dyn_connector($domain, $client_id, $client_secret,
			array($this, 'get_access_token'), array($this, 'set_access_token')
		);

		// Enable debugging output (should be disabled in production)
		$connector->set_debug(true);

		$data = array(
			'test' => array(
				'key1' => 1,
				'key2' => true,
				'key3' => 'This is a test'
			)
		);

		$result = $connector->send('auth/test/echo', $data);

		echo "Server response: \n";
		print_r($result);
	}

}



echo "<pre>";

$test = new test();

$test->normal_request();

echo "\n\n";

$test->authenticated_request();

echo "</pre>";

File: go/test/test_m2m.go
package main

import (
	"fmt"

	"modulo.srl/DynConnector/connector"
)

// Session persistor
type session struct {
	sessionToken string
}

func (s *session) GetSessionToken() string {
	fmt.Println("Connector requires session token (\"" + s.sessionToken + "\")")
	return s.sessionToken
}

func (s *session) SetSessionToken(token string) {
	s.sessionToken = token
	fmt.Println("New session token \"" + s.sessionToken + "\"")
}

// Test data to send and receive
type testData struct {
	KeyA    int64
	SubData testSubData
}
type testSubData struct {
	Key1 int
	Key2 bool
	Key3 string
}

func main() {
	domain := "modulo.srl"

	authUID := "test"
	masterToken := "test"

	session := session{}
	connector := connector.NewConnector(domain, authUID, masterToken, &session)

	// Enable debugging output (please disable in production)
	connector.SetDebug(true, nil, false)

	dataSend := testData{
		KeyA: 1024,
		SubData: testSubData{
			Key1: 1,
			Key2: true,
			Key3: "This is a test",
		},
	}
	dataReceive := testData{}

	/*dataSend := map[string]interface{}{
		"KeyA": 1024,
		"SubData": map[string]interface{}{
			"Key1": 1,
			"Key2": true,
			"Key3": "This is a test",
		},
	}
	dataReceive := map[string]interface{}{}
	*/

	err := connector.Send("echo", dataSend, &dataReceive)
	//err := connector.Send("echo/auth", dataSend, &dataReceive)

	if err != nil {
		fmt.Println("Error:", err.Error())
		return
	}

	fmt.Println("Server response:", dataReceive)
}
